### PR TITLE
ft. My highlights analytics - MH popup

### PR DESCRIPTION
**Pull request by KarinaMendez2**
for: https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/813

### CODE REVIEW

File: src/app/content/highlights/components/HighlightsPopUp.tsx [hunk 1]
```diff
@@ -24,16 +26,30 @@ interface Props {
   loginLink: string;
 }
 
-class HighlightsPopUp extends Component<Props> {
-  public popUp = React.createRef<HTMLElement>();
+// tslint:disable-next-line: variable-name
+const HighlightsPopUp = ({ ...props }: Props) => {
+  const popUpRef = React.createRef<HTMLElement>();
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/components/HighlightsPopUp.tsx]: `useRef`

File: src/app/content/highlights/components/SummaryPopup/HighlightListElement.tsx [hunk 2]
```diff
@@ -82,16 +83,23 @@ const HighlightListElement = ({ highlight, locationFilterId, pageId }: Highlight
...
   const dispatch = useDispatch();
 
+  const trackEditNoteColor = useAnalyticsEvent('editNoteColor');
+  const trackEditAnnotation = useAnalyticsEvent('editAnnotation');
+  const trackDeleteHighlight = useAnalyticsEvent('deleteHighlight');
+
   const updateAnnotation = (
     annotation: string
   ) => {
+    const addedNote = (highlight.annotation === undefined) ? true : false;
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/components/SummaryPopup/HighlightListElement.tsx]: `highlight.annotation === undefined` will already return `true` or `false` so ternary is not required

File: src/helpers/analyticsEvents/highlighting/summaryPopup/openClose.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { createSelector } from 'reselect';
+import * as selectNavigation from '../../../../app/navigation/selectors';
+import { AnalyticsEvent } from '../../event';
+
+const open = 'REX highlighting (open summary popup)';
+const close = 'REX highlighting (close summary popup)';
```
Comment (PiotrKozlowski) [inline src/helpers/analyticsEvents/highlighting/summaryPopup/openClose.ts]: Didn't check if this was specified in the issue, but in other events, we are using `MH popup`

**Review by PiotrKozlowski — COMMENTED** [review]
I've tried to add comments only to things I think are not related to <kbd>WIP</kbd>
Overall looks good to me

**Comment by PiotrKozlowski** [comment]
Changes look good. Should I make one more full review or wait until <kbd>WIP</kbd> label is removed?

**Comment by KarinaMendez2** [comment]
@PiotrKozlowski you can review, thanks

File: src/app/content/components/Toolbar/HighlightButton.tsx [hunk 4]
```diff
@@ -37,24 +38,31 @@ const MyHighlightsText = styled.span`
...
+  };
+
+  return <FormattedMessage id='i18n:toolbar:highlights:text'>
+      {(msg: Element | string) =>
+        <MyHighlightsWrapper onClick={() => openHighlightsSummary()} aria-label={msg}>
+          <MyHighlightsIcon aria-hidden='true' src={highlightIcon} />
+          <MyHighlightsText>{msg}</MyHighlightsText>
+        </MyHighlightsWrapper>
+      }
+    </FormattedMessage>
```
Comment (PiotrKozlowski) [inline src/app/content/components/Toolbar/HighlightButton.tsx]: This seems like too much indentation

File: src/app/content/highlights/components/HighlightsPopUp.spec.tsx [hunk 5]
```diff
@@ -98,10 +109,11 @@ describe('MyHighlights button and PopUp', () => {
     </Provider>, {createNodeMock});
 
     const isHtmlElement = jest.spyOn(appGuards, 'isHtmlElement');
-
     isHtmlElement.mockReturnValueOnce(true);
 
     act(() => { store.dispatch(openMyHighlights()); });
+    // Force componentDidUpdate()
+    act(() => { store.dispatch(receiveUser(user)); });
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/components/HighlightsPopUp.spec.tsx]: `HighlightsPopup` is no longer class component so probably this store update can be merged with the previous one

**Review by PiotrKozlowski — APPROVED** [review]
only some minor comments, overall looks good

**Pull request merged**